lib/meadow/pipeline/actions/common.ex

```diff
@@ -20,16 +20,23 @@ defmodule Meadow.Pipeline.Actions.Common do
         :ok
       end
 
-      def update_progress(attrs, {status, _, _}, _), do: update_progress(attrs, status)
-      def update_progress(attrs, {status, _}, _), do: update_progress(attrs, status)
-      def update_progress(attrs, {status, _}), do: update_progress(attrs, status)
+      def update_progress(data, attrs, {:error, error}) do
+        Progress.abort_remaining_pending_entries(attrs)
+        ActionStates.abort_remaining_waiting_actions(data.file_set_id)
+      end
+
+      def update_progress(data, attrs, {status, _, _}, _),
+        do: update_progress(data, attrs, status)
+
+      def update_progress(data, attrs, {status, _}, _), do: update_progress(data, attrs, status)
+      def update_progress(data, attrs, {status, _}), do: update_progress(data, attrs, status)
 
-      def update_progress(%{ingest_sheet: sheet_id, ingest_sheet_row: row_num}, status) do
+      def update_progress(data, %{ingest_sheet: sheet_id, ingest_sheet_row: row_num}, status) do
```

Comment:
I'm not getting a compiler warning about it (probably because the compiler isn't great at catching things in macros), it looks like `data` can be underscored out here.